### PR TITLE
Fix bugs with usage to scallops in filo cli

### PR DESCRIPTION
**Pull request by whizkido**
**Pull Request checklist**

- [x] The commit(s) message(s) follows the contribution [guidelines](CONTRIBUTING.md) ?
- [x] Tests for the changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?

**Current behavior :** (link exiting issues here : https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests)

There were three issues which are fixed as part of this pr

1.  Fixed the org.rogach.scallop.exceptions.IncompleteBuildException
2.  Unknown option error caused because options were not in small case
3.   start and end were hardcoded and was not using the param being passed.

**New behavior :**

Fixed the above issues.
All options need to be passed in small case. I will update the document as part of separate PR.
Added a test class to test the parsing of command line options.

### CODE REVIEW

File: cli/src/main/scala/filodb.cli/CliMain.scala [hunk 1]
```diff
@@ -25,41 +23,50 @@ import filodb.memory.format.{BinaryVector, Classes, MemoryReader, RowReader}
...
   val minutes = opt[String]()
   val step = opt[Long](default = Some(10)) // in seconds
   val chunks = opt[String]()   // select either "memory" or "buffers" chunks only
-  val everyNSeconds = opt[String]()
+  val everynseconds = opt[String]()
   val shards = opt[List[String]]()
   val spread = opt[Int]()
   verify()
+
+  override def onError(e: Throwable): Unit = e match {
```
Comment (broneill) [inline cli/src/main/scala/filodb.cli/CliMain.scala:64]: Why? You always throw the exception back out.

File: cli/src/main/scala/filodb.cli/CliMain.scala [hunk 2]
```diff
@@ -25,41 +23,50 @@ import filodb.memory.format.{BinaryVector, Classes, MemoryReader, RowReader}
...
+  val hexchunkinfo = opt[String]()
+  val vectortype = opt[String]()
   val matcher = opt[String]()
-  val labelNames = opt[List[String]](default = Some(List()))
-  val labelFilter = opt[Map[String, String]](default = Some(Map.empty))
-  val start: Long = System.currentTimeMillis() / 1000 // promql argument is seconds since epoch
-  val end: Long = System.currentTimeMillis() / 1000 // promql argument is seconds since epoch
+  val labelnames = opt[List[String]](default = Some(List()))
+  val labelfilter = opt[Map[String, String]](default = Some(Map.empty))
+  val start = opt[BigInt](default = Some(System.currentTimeMillis()/1000))// promql argument is seconds since epoch
```
Comment (broneill) [inline cli/src/main/scala/filodb.cli/CliMain.scala:54]: Long isn't big enough?

File: cli/src/main/scala/filodb.cli/CliMain.scala [hunk 3]
```diff
@@ -165,39 +173,40 @@ object CliMain extends FilodbClusterNode {
...
         case Some("decodeChunkInfo") =>
-          require(args.hexChunkInfo.isDefined, "--hexChunkInfo must be defined")
-          decodeChunkInfo(args.hexChunkInfo())
+          require(args.hexchunkinfo.isDefined, "--hexChunkInfo must be defined")
+          decodeChunkInfo(args.hexchunkinfo())
 
         case Some("decodeVector") =>
-          require(args.hexVector.isDefined && args.vectorType.isDefined, "--hexVector and --vectorType must be defined")
-          decodeVector(args.hexVector(), args.vectorType())
+          println("inside decodeVector "+args.hexvector)
```
Comment (broneill) [inline cli/src/main/scala/filodb.cli/CliMain.scala:184]: println? Leftover debug?

File: cli/src/main/scala/filodb.cli/CliMain.scala (hunk 2 above)
Comment (whizkido) [reply cli/src/main/scala/filodb.cli/CliMain.scala]: Fixed this in latest commit.

File: cli/src/main/scala/filodb.cli/CliMain.scala (hunk 1 above)
Comment (whizkido) [reply cli/src/main/scala/filodb.cli/CliMain.scala:69]: This is needed for using intersect in the test spec. else the error returned is a different string  for different errors.

File: cli/src/main/scala/filodb.cli/CliMain.scala (hunk 3 above)
Comment (whizkido) [reply cli/src/main/scala/filodb.cli/CliMain.scala]: Yes this and one other println was left behind. Fixed it.

**Review by vishramachandran — APPROVED** [review]

**Pull request merged**